dev/com.ibm.ws.cdi.1.2.weld/src/com/ibm/ws/cdi/proxy/ProxyServicesImpl.java

```diff
@@ -53,6 +53,14 @@ public void cleanup() {
 
     @Override
     public ClassLoader getClassLoader(final Class<?> proxiedBeanType) {
+
+        if (System.getSecurityManager() != null) {
+            String className = new Exception().getStackTrace()[1].getClassName();
+            if (! className.equals("org.jboss.weld.bean.proxy.ProxyFactory")) {
```

Comment:
I think this class is exported so we should be able to use the class directly rather than its name?

Reply:
StackTraceElement doesn't have any reference to the Class, just a string for it's name. I could get the name from ProxyFactory.class but that seems unnecessary.

Reply:
Ok, it doesn't matter in this case since tests will fail if the name is wrong.